src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * duke.Deadline is a duke.Task that accepts a task description and a deadline (dd-MM-yyyy)
+ */
+public class Deadline extends Task {
+
+    protected String by;
+    protected LocalDate byDate;
+
+
+    /**
+     * Constructor for duke.Deadline
+     *
+     * @param description description of the task
+     * @param by          date in the format of dd-MM-yyyy
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.byDate = LocalDate.parse(by, DateTimeFormatter.ofPattern("dd-MM-yyyy"));
```

Comment:
DateTimeFormatter.ofPattern("dd-MM-yyyy") can be declared as a static variable. I think it can reusable in the scenario that more formatting is needed to be done?